### PR TITLE
Support setOrder

### LINKED ISSUE #1735: Add support for `Order`
Related to https://github.com/hibernate/hibernate-reactive/issues/1706

See the new class in ORM: [org.hibernate.query.Order](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/query/Order.java)

### PR DESCRIPTION
**Pull request by DavideD**
fix #1735 
Supersedes #1813

### CODE REVIEW

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/OrderTest.java
Comment (blafond) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/OrderTest.java:1]: this import isn't used

`import static org.junit.jupiter.api.Assertions.assertEquals;`

**Review by blafond — APPROVED** [review]
Just one change... the test has an unused import:

import static org.junit.jupiter.api.Assertions.assertEquals;

**Comment by DavideD** [comment]
Thanks, there also something else that doesn't work. I'm looking into it

**Comment by blafond** [comment]
I tested locally and there are some failed tests :(

**Comment by DavideD** [comment]
Should be fixed now, let's see what CI think about it

**Pull request merged**